portfolio/src/main/webapp/script-map.js

```diff
@@ -0,0 +1,17 @@
+/*
+ * Similar to an inline script approach without including initMap within the html
+ * itself, creates a map.
+ */
+const script = document.createElement('script');
```

Comment:
Refrain from putting code in the global scope. It becomes really hard to know what code is running. 
Wrap all of this in a function called initMap() and call initMap() in your main javascript script. Wherever the main() function is.